### PR TITLE
qt5: 5.15.0 -> 5.15.2

### PR DESCRIPTION
**Pull request by andir**
###### Motivation for this change

See commit messages.

###### Things done

- [X] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [X] NixOS
- [X] executed cutebrowser with the new Qt build.
- [X] executed the `plasma5` NixOS test.

### CODE REVIEW

**Comment by andir** [comment]
I'm still building the `plasma5` test locally to identify the blast radius of this change. If it wouldn't be for that QtWebEngine crash I'd retarget to staging.

File: pkgs/development/libraries/qt-5/5.15/default.nix [hunk 1]
```diff
@@ -79,13 +79,6 @@ let
     qtserialport = [ ./qtserialport.patch ];
     qtwebengine = [
       # Fix build with bison-3.7: https://code.qt.io/cgit/qt/qtwebengine-chromium.git/commit/?id=1a53f599
```
Comment (veprbl) [inline pkgs/development/libraries/qt-5/5.15/default.nix:81]: This comment needs to be removed as well

Comment (andir) [reply pkgs/development/libraries/qt-5/5.15/default.nix]: removed

**Comment by FRidh** [comment]
Rebuilding my desktop against this now.

**Comment by FRidh** [comment]
```
$ nix build .#kompare
...
[ 85%] Building CXX object komparepart/CMakeFiles/komparepart.dir/komparelistview.cpp.o
[ 87%] Building CXX object komparepart/CMakeFiles/komparepart.dir/kompareprefdlg.cpp.o
/build/kompare-20.08.2/komparepart/kompare_part.cpp: In member function 'void KomparePart::slotFilePrint()':
/build/kompare-20.08.2/komparepart/kompare_part.cpp:613:13: error: 'class QPrinter' has no member named 'setOrientation'; did you mean 'Orientation'?
  613 |     printer.setOrientation(QPrinter::Landscape);
      |             ^~~~~~~~~~~~~~
      |             Orientation
/build/kompare-20.08.2/komparepart/kompare_part.cpp: In member function 'void KomparePart::slotFilePrintPreview()':
/build/kompare-20.08.2/komparepart/kompare_part.cpp:628:13: error: 'class QPrinter' has no member named 'setOrientation'; did you mean 'Orientation'?
  628 |     printer.setOrientation(QPrinter::Landscape);
      |             ^~~~~~~~~~~~~~
      |             Orientation
[ 89%] Building CXX object komparepart/CMakeFiles/komparepart.dir/komparesaveoptionsbase.cpp.o
make[2]: *** [komparepart/CMakeFiles/komparepart.dir/build.make:105: komparepart/CMakeFiles/komparepart.dir/kompare_part.cpp.o] Error 1
make[2]: *** Waiting for unfinished jobs....
[ 91%] Linking CXX executable kompare
[ 91%] Built target kompare
make[1]: *** [CMakeFiles/Makefile2:926: komparepart/CMakeFiles/komparepart.dir/all] Error 2
make: *** [Makefile:149: all] Error 2
```

**Comment by FRidh** [comment]
See also https://github.com/NixOS/nixpkgs/pull/102216 and https://github.com/NixOS/nixpkgs/pull/104448

**Comment by andir** [comment]
Oh wow, the ShitHub search hasn't brought up anything useful when I
searched....

**Comment by unclechu** [comment]
Happy to see that there’s activity here and all the different MRs are shrinked into this only one. Hopefully my eyes will be saved from all that bright light soon. :smile:

**Comment by ttuegel** [comment]
I have re-built `qt515` and the only failure was `qtwebkit`, which was failing before. At this point, I think any other failures can be fixed in the staging stabilization period. Unless there is any objection from @NixOS/qt-kde, I will rebase this onto staging and merge. (I will wait at least 24 hours for objections, but I feel like the risk is low because it has to go into staging.)

**Comment by doronbehar** [comment]
> I have re-built `qt515` and the only failure was `qtwebkit`, which was failing before.

@ttuegel please see https://github.com/NixOS/nixpkgs/pull/104474 .

**Comment by FRidh** [comment]
@petabyteboy the maintainer wrote to wait with merging

**Comment by FRidh** [comment]
https://invent.kde.org/sdk/kompare/-/commit/68d3eee36c48a2f44ccfd3f9e5a36311b829104b is needed for kompare

**Comment by petabyteboy** [comment]
> @petabyteboy the maintainer wrote to wait with merging
> 
> (oh wait we're maintainers as well)

I didn't read it carefully anough and overlooked the "24 hours" part, sorry about that.

Should we put the qtwebkit fix to staging as well to prevent any merge conflicts later or put it on master because it's not a mass rebuild and not directly related to this change?

**Comment by ttuegel** [comment]
> I didn't read it carefully enough and overlooked the "24 hours" part, sorry about that.

Don't worry about it, I was mostly waiting to see if anyone would point out some problem I hadn't noticed, and most of the people who would do that have already commented here. :sweat_smile: 

> Should we put the qtwebkit fix to staging as well to prevent any merge conflicts later or put it on master because it's not a mass rebuild and not directly related to this change?

I think we should go ahead and merge the qtwebkit changes to master. There are many packages which can probably be upgraded from Qt 5.14 to Qt 5.15 now, so it would be good if those package maintainers could start testing as soon as possible.

**Comment by unclechu** [comment]
Any ideas when I could test it in 20.09 release?

**Comment by poelzi** [comment]
same question. 20.09 is in it's current state just garbage and unusable by any means. One of my main programs I use for performances crashes regularly because of this.

**Comment by ttuegel** [comment]
I opened #107713 to port this update to NixOS 20.09. Please note that this change must go through the staging branch, so it may take several weeks to reach the release channel.

**Comment by unclechu** [comment]
This is now delivered to the `nixos-20.09` release branch. `vlc`, `smplayer`, `audacious`, `audacity` are all using dark theme now.
But only `psi-plus` for me is still not using dark theme. I checked and discovered that it still linked to Qt 5.14.x. Should it be?

**Pull request merged**